### PR TITLE
Release 1.0.8: Detect mints when paying with non-ETH tokens

### PR DESCRIPTION
**Pull request by guscost**
Resolves OU-2124

### CODE REVIEW

**Review by pcowgill — CHANGES_REQUESTED** [review]
@guscost Can you please include a test that was failing before and passes with these changes? You can use the `npm run grab:transaction [txHash] [txNickname]` command to grab one to use in the tests.

**Review by ponyjackal — APPROVED** [review]

File: src/contextualizers/heuristics/erc1155Mint/erc1155Mint.spec.ts [hunk 1]
```diff
@@ -33,6 +34,13 @@ describe('ERC1155 Mint', () => {
     expect(erc1155Mint2).toBe(true);
   });
 
+  it('Should detect ERC1155 Mint transaction when paid in a non-ETH token', () => {
+    const erc1155Mint3 = detect(
+      erc1155MintPaidWithEnjoy as unknown as Transaction,
+    );
+    expect(erc1155Mint3).toBe(true);
+  });
+
```
Comment (pcowgill) [inline src/contextualizers/heuristics/erc1155Mint/erc1155Mint.spec.ts:43]: @guscost to add generate test case using the contextSummary helper like in below tests

Comment (guscost) [reply src/contextualizers/heuristics/erc1155Mint/erc1155Mint.spec.ts:43]: Done!

File: src/contextualizers/heuristics/erc1155Mint/erc1155Mint.spec.ts [hunk 2]
```diff
@@ -61,4 +61,15 @@ describe('ERC1155 Mint', () => {
...
   });
+
+  it('Should generate a context for ERC1155 Mint paid with another token', () => {
+    const erc1155Mint3 = generate(
+      erc1155MintPaidWithEnjoy as unknown as Transaction,
+    );
+    expect(erc1155Mint3.context?.summaries?.category).toBe('NFT');
+    const desc3 = contextSummary(erc1155Mint3.context);
+    expect(desc3).toBe(
+      '0xf1156f02a3f50ab64bce6d637d36c0b77ec1d5a5 MINTED 113 0x634d2e86b78a69fbe91104b6077c0aa403867558 #14 for 113 ETH',
```
Comment (pcowgill) [inline src/contextualizers/heuristics/erc1155Mint/erc1155Mint.spec.ts:72]: @guscost Shouldn't it not say ETH here, if they're paying with another token?

**Review by pcowgill — CHANGES_REQUESTED** [review]
https://github.com/Once-Upon/context/pull/302/commits/f2ca6ce6f89ed0b413e414bd66f581d8258a5c2f#r1561250179

File: src/contextualizers/heuristics/erc1155Mint/erc1155Mint.spec.ts (hunk 2 above)
Comment (guscost) [reply src/contextualizers/heuristics/erc1155Mint/erc1155Mint.spec.ts:72]: Ohhhh

File: src/contextualizers/heuristics/erc1155Mint/erc1155Mint.ts [hunk 3]
```diff
@@ -79,12 +81,24 @@ export function generate(transaction: Transaction): Transaction {
   const recipient = assetTransfer.to;
   const amount = mints.filter((ele) => ele.type === assetTransfer.type).length;
 
-  const assetSent = transaction.netAssetTransfers[transaction.from]
-    ?.sent as ETHAsset[];
+  const assetSent = transaction.netAssetTransfers[transaction.from]?.sent as (
+    | ETHAsset
```
Comment (pcowgill) [inline src/contextualizers/heuristics/erc1155Mint/erc1155Mint.ts:85]: Anywhere we have ETHAsset, we should have DEGENAsset too

**Review by pcowgill — APPROVED** [review]

File: src/contextualizers/heuristics/erc1155Mint/erc1155Mint.ts [hunk 4]
```diff
@@ -79,12 +81,24 @@ export function generate(transaction: Transaction): Transaction {
...
+    | ERC20Asset
+  )[];
+
   const price =
-    assetSent && assetSent.length > 0 && assetSent[0]?.value
-      ? assetSent[0].value
-      : '0';
+    assetSent && assetSent.length > 0 && assetSent[0].type === 'erc20'
```
Comment (pcowgill) [inline src/contextualizers/heuristics/erc1155Mint/erc1155Mint.ts:90]: @guscost Do we need to make this same change in the ERC721 and ERC20 versions?

**Pull request merged**